samples/net/irc_bot/src/irc-bot.c

```diff
@@ -834,239 +632,39 @@ on_msg_rcvd(void *data, struct zirc_chan *chan, char *umask, char *msg)
-	}
-
-	k_sem_give(&irc->wait_iface);
-}
-
-#else
-#define setup_ipv4(...)
-#endif /* CONFIG_NET_IPV4 && !CONFIG_NET_DHCPV4 */
-
-#if defined(CONFIG_NET_IPV6)
-
-#define DAD_TIMEOUT K_SECONDS(3)
-
-static struct net_mgmt_event_callback mgmt6_cb;
-static K_SEM_DEFINE(dad_ok, 0, UINT_MAX);
-static struct in6_addr laddr;
-
-static void ipv6_dad_ok_handler(struct net_mgmt_event_callback *cb,
-				u32_t mgmt_event,
-				struct net_if *iface)
-{
-	struct net_if_addr *ifaddr;
-
-	if (mgmt_event != NET_EVENT_IPV6_DAD_SUCCEED) {
-		return;
-	}
-
-	ifaddr = net_if_ipv6_addr_lookup(&laddr, &iface);
-	if (!ifaddr ||
-	    !(net_ipv6_addr_cmp(&ifaddr->address.in6_addr, &laddr) &&
-	      ifaddr->addr_state == NET_ADDR_PREFERRED)) {
-		/* Address is not yet properly setup */
-		return;
-	}
-
-	k_sem_give(&dad_ok);
-}
-
-static void setup_ipv6(struct zirc *irc, struct net_if *iface)
-{
-#if CONFIG_SYS_LOG_NET_LEVEL > 2
-	char hr_addr[NET_IPV6_ADDR_LEN];
-#endif
-
-	if (net_addr_pton(AF_INET6, ZIRC_LOCAL_IP_ADDR, &laddr)) {
-		NET_ERR("Invalid address: %s", ZIRC_LOCAL_IP_ADDR);
-		return;
-	}
-
-	NET_INFO("IPv6 address: %s",
-		 net_addr_ntop(AF_INET6, &laddr, hr_addr, NET_IPV6_ADDR_LEN));
-
-	net_mgmt_init_event_callback(&mgmt6_cb, ipv6_dad_ok_handler,
-				     NET_EVENT_IPV6_DAD_SUCCEED);
-	net_mgmt_add_event_callback(&mgmt6_cb);
-
-	net_if_ipv6_addr_add(iface, &laddr, NET_ADDR_MANUAL, 0);
-
-	if (k_sem_take(&dad_ok, DAD_TIMEOUT) < 0) {
-		panic("IPv6 address setup failed.");
-	}
-
-	if (in_addr_set(ZIRC_AF_INET, ZIRC_LOCAL_IP_ADDR, 0,
-			&irc->local_addr) < 0) {
-		NET_ERR("Invalid IP address: %s", ZIRC_LOCAL_IP_ADDR);
-	}
-
-	k_sem_give(&irc->wait_iface);
-}
-
-#else
-#define setup_ipv6(...)
-#endif /* CONFIG_NET_IPV6 */
-
-static void
-initialize_network(struct zirc *irc)
+void main(void)
 {
-	struct net_if *iface;
-
-	NET_INFO("Initializing network");
-
-#if defined(CONFIG_DNS_RESOLVER)
-	k_sem_init(&irc->wait_dns, 0, UINT_MAX);
-#endif
-
-	k_sem_init(&irc->wait_iface, 0, UINT_MAX);
-
-	iface = net_if_get_default();
-	if (!iface) {
-		panic("No default network interface");
-	}
-
-#if defined(CONFIG_NET_IPV6) && defined(CONFIG_NET_DHCPV6)
-	/* TODO: IPV6 DHCP */
-#elif !defined(CONFIG_NET_IPV6) && defined(CONFIG_NET_DHCPV4)
-	setup_dhcpv4(irc, iface);
-#elif !defined(CONFIG_NET_IPV6) && defined(CONFIG_NET_IPV4)
-	setup_ipv4(irc, iface);
-#else
-	setup_ipv6(irc, iface);
-#endif
-
-	/* Wait until we are ready to continue */
-	k_sem_take(&irc->wait_iface, K_FOREVER);
-}
+	int ret;
 
-static void
-initialize_hardware(void)
-{
-	NET_INFO("Initializing hardware");
+	SYS_LOG_INF(APP_BANNER);
 
 	led0 = device_get_binding(LED_GPIO_NAME);
 	if (led0) {
 		gpio_pin_configure(led0, LED_PIN, GPIO_DIR_OUT);
 	}
-}
 
-void main(void)
-{
-	struct zirc irc = { };
-	struct zirc_chan chan = { };
-	int ret;
-
-	NET_INFO("Zephyr IRC bot sample");
+	memset(&app_ctx, 0, sizeof(app_ctx));
+	memset(cmd_buf, 0, CMD_BUFFER_SIZE);
```

Comment:
These two are zeroed out as part of clearing the BSS, no need to memset() them here.

Reply:
Ugh, I had these as parameters and moved them out as statics later.  Another good catch.